dijet/tasks/root.py

```diff
@@ -0,0 +1,104 @@
+    It is not yet possible to write TGraphs via uproot.
+    Efforts by the uproot Team ongoing. Related PR:
+    https://github.com/scikit-hep/uproot5/pull/1144
+    """
+
+    output_collection_cls = law.NestedSiblingFileCollection
+
+    # upstream requirements
+    reqs = Requirements(
+        JER=JER,
+    )
+
+    # TODO: write base task for root conversion
+    #       keep HistogramBaseTask as long as no conversion possible
+    def requires(self):
+        return self.reqs.JER.req(
+            self,
+            branch=-1,
+            _exclude={"branches"},
+        )
+
+    def workflow_requires(self):
+        reqs = super().workflow_requires()
+        reqs["key"] = self.as_branch().requires()
+        return reqs
+
+    def load_jer(self):
+        histogram = self.input().collection[0]["jers"].load(formatter="pickle")
+        return histogram
+
+    def output(self) -> dict[law.FileSystemTarget]:
+        sample = self.extract_sample()
+        target = self.target(f"{sample}", dir=True)
+        outp = {
+            "jers": target.child("jers.pickle", type="f"),
+        }
+        return outp
+
+    def run(self):
+        results_jer = self.load_jer()
+        jer = results_jer["jer"]
+
+        eta_bins = jer.axes["probejet_abseta"].edges
+
+        # Get error for pt bins. These will be symmetrical for the moment
+        # TODO: Define pt value by the mean value of the pt in given a pt bin
+        pt_bins = jer.axes["dijets_pt_avg"].edges
+        pt_centers = jer.axes["dijets_pt_avg"].centers
+        pt_error_low = pt_centers - pt_bins[:-1]
+        pt_error_high = pt_bins[1:] - pt_centers
+
+        def get_eta_bins(self, i: int):
+            up = "{:.3f}".format(eta_bins[i])
+            do = "{:.3f}".format(eta_bins[i + 1])
+            bin_low = up.replace(".", "p")
+            bin_high = do.replace(".", "p")
+            return bin_low, bin_high
+
+        # Store values for TGraphAsymmError in dictionary and convert with additional script.
+        results_jers = {}
+        for m in self.category_id:
+            for i in range(len(eta_bins) - 1):
+                low, high = get_eta_bins(self, i)
+                tmp = jer[
+                    {
+                        "category": hist.loc(self.category_id[m]),
+                        "probejet_abseta": i,
+                    }
+                ]
+                jers = tmp.values()
+                jers_errors = tmp.variances()
+                assert len(jers) == len(pt_centers), f"Check number of bins for {i}"
+                results_jers[f"jer_eta_{low}_{high}_{m}"] = {
+                    "fY": jers,
+                    "fYerrUp": jers_errors,
+                    "fYerrDown": jers_errors,
+                    "fX": pt_centers,
+                    "fXerrUp": pt_error_high,
+                    "fXerrDown": pt_error_low,
+                }
+
+        self.output()["jers"].dump(results_jers, formatter="pickle")
```

Comment:
A few things here:
* since `jer` is a histogram, it would be good to reflect this in the variable name -> `h_jer`
* the absolute pseudorapidity should be consistently called `abseta`, not just `eta`
* having a function to compute the bin names seems overkill, better to do it inline in the loop
* using f-strings should be preferred to `str.format`
* one can loop over the bin edges in pairs using `zip`, which is cleaner IMO than using the index
* accessing the abs(eta) bin by index also seems error-prone, one could use `hist.loc` instead

Here's an implementation suggestion taking care of the above:
```suggestion
    def run(self):
        results_jer = self.load_jer()
        h_jer = results_jer["jer"]

        # get pt bins (errors symmetrical for now)
        pt_bins = h_jer.axes["dijets_pt_avg"].edges
        pt_centers = h_jer.axes["dijets_pt_avg"].centers
        pt_error_lo = pt_centers - pt_bins[:-1]
        pt_error_hi = pt_bins[1:] - pt_centers

        # compute and store values for building `TGraphAsymmError`
        # in external script
        results_jers = {}
        abseta_bins = h_jer.axes["probejet_abseta"].edges
        for method in self.LOOKUP_CATEGORY_ID:
            for abseta_lo, abseta_hi in zip(
                abseta_bins[:-1],
                abseta_bins[1:],
            ):
                 
                abseta_lo_str = f"{abseta_lo:.3f}".replace(".", "p")
                abseta_hi_str = f"{abseta_hi:.3f}".replace(".", "p")
                abseta_str = f"abseta_{abseta_lo_str}_{abseta_hi_str}"

                abseta_center = (abseta_lo + abseta_hi) / 2
                h_jer_category_abseta = h_jer[
                    {
                        "category": hist.loc(self.LOOKUP_CATEGORY_ID[method]),
                        "probejet_abseta": hist.loc(abseta_center),
                    }
                ]

                jers = h_jer_category_abseta.values()
                jers_errors = h_jer_category_abseta.variances()

                # sanity check
                assert len(jers) == len(pt_centers), f"check number of bins for {abseta_str!r}"

                # store results
                results_jers[f"jer_{abseta_str}_{method}"] = {
                    "fY": jers,
                    "fYerrUp": jers_errors,
                    "fYerrDown": jers_errors,
                    "fX": pt_centers,
                    "fXerrUp": pt_error_hi,
                    "fXerrDown": pt_error_lo,
                }

        # store results
        self.output()["jers"].dump(results_jers, formatter="pickle")
```